lib/formats/dotnet-core-parser.js

```diff
@@ -1,82 +1,153 @@
+  freqDeps['System.Runtime'] = false;
+  freqDeps['System.IO'] = false;
+  freqDeps['System.Text.Encoding'] = false;
+  freqDeps['System.Threading.Tasks'] = false;
+  freqDeps['System.Reflection'] = false;
+  freqDeps['System.Globalization'] = false;
+  freqDeps.dependencies = {};  
+}
+
 function convertFromPathSyntax(path) {
-  var name = path.split('/').join('@'); // posix
+  let name = path.split('/').join('@'); // posix
   name = name.split('\\').join('@'); // windows
   return name;
 }
 
 function collectFlatList(targetObj) {
-  var names = Object.keys(targetObj);
-  var libs = names.map(function (name) {
+  const names = Object.keys(targetObj);
+  return names.map(function (name) {
     name = convertFromPathSyntax(name);
     return name;
   });
-  return libs;
 }
 
-function registerDependency(packageTree, resolvedName) {
-  var name = resolvedName.split('@')[0];
-  var version = resolvedName.split('@')[1];
-  packageTree.dependencies[name] =
-    packageTree.dependencies[name] || new Dependency(name, version);
-  debug(resolvedName);
+function buildTreeRecursive(targetDeps, depName, parent, treeDepth) {
+  if (treeDepth > 20)
+  {
+    throw new Error('The depth of the tree is too big.');
+  }
+  
+  let depResolvedName = "";
+  let originalDepKey = "";
+  
+  debug(`${treeDepth}: Looking for '${depName}'`);
+  const exists = Object.keys(targetDeps).some(function (currentDep) {
+      let currentResolvedName = convertFromPathSyntax(currentDep);
+      if(currentResolvedName.split('@')[0].toLowerCase() === depName){
+        depResolvedName = currentResolvedName;
+        originalDepKey = currentDep;
+        debug(`${treeDepth}: Found '${currentDep}'`);
+        return true;
+      }
+  });
+
+  if (!exists) {
+    debug(`Failed to find '${depName}'`);
+    return;
+  }
+
+  const depVersion = depResolvedName.split('@')[1];
+
+  parent.dependencies[depName] = parent.dependencies[depName] || new Dependency(depName, depVersion);
+
+  Object.keys(targetDeps[originalDepKey].dependencies || {}).forEach(function (currentDep)
+  {
+    if (currentDep in freqDeps)
+    {
+      if (freqDeps[currentDep] != false)
+      {
+        return;
+      }
+
+      buildTreeRecursive(targetDeps, currentDep.toLowerCase(), freqDeps, 0);
+      freqDeps[currentDep] = true;
+    }
+    else
+    {
+      buildTreeRecursive(targetDeps, currentDep.toLowerCase(), parent.dependencies[depName], treeDepth + 1);
+    }
+  });
+}
+
+
+function getFrameworkObjToRun(manifest) {
+  if (!manifest.project.frameworks) {
+    throw new Error('No frameworks were found in project.assets.json');
+  }
+
+  const frameworks = manifest.project.frameworks;
+  if (Object.keys(frameworks).length == 0) {
+    throw new Error('0 frameworks were found in project.assets.json');
+  }
+
+  debug(`Available frameworks: '${Object.keys(frameworks)}'`);
+  
+  //not yet supporting multiple frameworks in the same assets file -> taking only the first 1
```

Comment:
What's the issue with supporting multiple frameworks in the assets file? It was already generated for us as expected, would we be removing dependencies if we don't adhere to the contents of `project.assets.json`

Reply:
Explained :)